cJSON.h

```diff
@@ -50,6 +50,7 @@ typedef struct cJSON {
 	char *valuestring;			/* The item's string, if type==cJSON_String */
 	int valueint;				/* The item's number, if type==cJSON_Number */
 	double valuedouble;			/* The item's number, if type==cJSON_Number */
+	unsigned long long valueull;       /* The item's number, if type==cJSON_Number */
```

Comment:
unsigned long long is a very special case. It's unsigned, so no negative numbers and the range of supported numbers is not supported by most JSON libraries out there.